File: Queue in data structures/queue_double_pointer.py
import numpy as np 

class queue() :
    
    def __init__(self,size) :
        self.size = size
        self.q = np.array([None]*self.size)
        self.front = self.rear = -1
        
    def isempty(self) :
        if self.front == self.rear :
            return True
        else :
            return False
        
    def isfull(self) :
        if self.rear == self.size-1 :
            return True
        else :
            return False 
        
    def enqueue(self,data) :
        if not self.isfull() :
            self.rear += 1
            self.q[self.rear] = data
        else :
            print("\nQueue is full cannot insert ",data,"\n")
            
    def dequeue(self) :
        if not self.isempty() :
            self.front += 1
            return self.q[self.front]
        else :
            print("\nQueue is empty cannot dequeue\n")
            
    def get_values(self) :
        if not self.isempty() :
            print("\nValues present in queue are :- ",end = " ")
            for x in range(self.front+1,self.rear+1) :
                print(self.q[x],end="\t")
            print()
        else :
            print("\nQueue is empty there are no values to print \n")
        
qu = queue(5)

qu.enqueue(2)
qu.enqueue(3)
qu.enqueue(5)
qu.get_values()
qu.dequeue()
qu.get_values()

File: Queue in data structures/queue_double_pointer.cpp
#include <iostream>
using namespace std;
class Queue
{
private:
int front;
int rear;
int size;
int *qu;
public:
Queue(){front=rear=-1;size=10;qu=new int[size];}
Queue(int size){front=rear=-1;
this -> size=size;qu=new int[this->size];}
void enqueue(int x);
int dequeue();
void Display();
};

void Queue::enqueue(int x)
{
if(rear==size-1)
printf("Queue Full\n");
else
{
rear++;
qu[rear]=x;
}
}

int Queue::dequeue()
{
int x=-1;
if(front==rear)
printf("Queue is Empty\n");
else

{
x=qu[front+1];
front++;
}
return x;
}

void Queue::Display()
{
printf("Vlaues present in queue are :- ");
for(int i=front+1;i<=rear;i++)
printf("%d ",qu[i]);
printf("\n\n");
}

int main()
{
Queue q(5);
q.enqueue(2);
q.enqueue(3);
q.enqueue(5);
q.Display();
q.dequeue();
q.Display();
return 0;
}